### PR TITLE
Get rid off dependency on Signal in Win32 Process::Status

### PR DESCRIPTION
**Pull request by jan-zajic**
Remove dependency on Signal on win32 platform.

### CODE REVIEW

File: src/compiler/crystal/compiler.cr [hunk 1]
```diff
@@ -577,7 +577,10 @@ module Crystal
       status = yield command, args
 
       unless status.success?
-        msg = status.normal_exit? ? "code: #{status.exit_code}" : "signal: #{status.exit_signal} (#{status.exit_signal.value})"
+        msg = "code: #{status.exit_code}"
+        {% unless flag?(:win32) %}
+          msg = status.normal_exit? ? "code: #{status.exit_code}" : "signal: #{status.exit_signal} (#{status.exit_signal.value})"
```
Comment (Sija) [inline src/compiler/crystal/compiler.cr]: Please, keep it dry:

```suggestion
          msg = "signal: #{status.exit_signal} (#{status.exit_signal.value})" unless status.normal_exit?
```

File: src/compiler/crystal/tools/playground/server.cr [hunk 2]
```diff
@@ -175,7 +175,10 @@ module Crystal::Playground
       spawn do
         status = process.wait
         @logger.info "Code execution ended (session=#{@session_key}, tag=#{tag}, filename=#{output_filename})."
-        exit_status = status.normal_exit? ? status.exit_code : status.exit_signal.value
+        exit_status = status.exit_code
+        {% unless flag?(:win32) %}
+          exit_status = status.normal_exit? ? status.exit_code : status.exit_signal.value
```
Comment (Sija) [inline src/compiler/crystal/tools/playground/server.cr]: ditto

```suggestion
          exit_status = status.exit_signal.value unless status.normal_exit?
```

**Comment by straight-shoota** [comment]
I'd like to see a comment for each location that is removed which explains why this doesn't work for win32, whether that's a problem and if this needs attention later.

**Comment by RX14** [comment]
Yeah, I'd much rather add something to the windows stubs file than ifdef out code. Ifdefs should be kept to a minimum.

**Comment by straight-shoota** [comment]
The result of #8647 is also relevant for this.

**Pull request closed without merging**